app/views/events/show.html.erb

```diff
@@ -30,8 +30,11 @@
 <% end %>
 
 <div id="join-event">
-  <% if current_user && current_user.invites.where(event_id: @event.id).count.zero? %>
-    <%= link_to "Join Event", "/invites?invite[status]=Accepted&invite[event_id]=#{@event.id}&invite[user_id]=#{current_user.id}", method: :POST, class: "btn btn-default" %>
+  <% if current_user && !@event.private && current_user.invites.where(event_id: @event.id).where("status ILIKE ?", 'Accepted%').count.zero? %>
```

Comment:
We should try to move all this logic and queries out of the view. This might be a good place for a helper method for the conditional and a method on user for the invite query.